### PR TITLE
Collapse drawer on home press

### PR DESCRIPTION
**Pull request by Cuiboy**

### CODE REVIEW

**Review by Stereopoly — APPROVED** [review]
Works well!

File: berkeley-mobile/Drawer/DrawerViewController.swift [hunk 1]
```diff
@@ -37,8 +39,16 @@ class DrawerViewController: UIViewController {
         
         setupBackgroundView()
         setupGestures()
+        
+        let nc = NotificationCenter.default
+        nc.addObserver(self, selector: #selector(collapseTopDrawer), name: Notification.Name(TabBarController.homePressedCollapseMessage), object: nil)
     }
     
+    @objc func collapseTopDrawer() {
+        delegate.moveDrawer(to: .collapsed, duration: 0.5)
```
Comment (huangshawn1) [inline berkeley-mobile/Drawer/DrawerViewController.swift:48]: small nitpick but should remove duration: 0.5 (just have delegate.moveDrawer(to: .collapsed))
default move duration is 0.5 anyway and we should keep all the default move durations in one place

File: berkeley-mobile/Drawer/DrawerViewController.swift [hunk 2]
```diff
@@ -11,6 +11,7 @@ import UIKit
 // General drawer with a gray bar at the top, can handle pan gesture to change position
 class DrawerViewController: UIViewController {
     var delegate: DrawerViewDelegate!
+    var mainDelegate: MainDrawerViewDelegate!
```
Comment (huangshawn1) [inline berkeley-mobile/Drawer/DrawerViewController.swift:14]: doesn't look like this is used anywhere, remove?

**Review by huangshawn1 — APPROVED** [review]
looks good/works well, couple minor things

**Pull request merged**